locator/network_topology_strategy.cc

```diff
@@ -407,22 +407,9 @@ network_topology_strategy::reallocate_tablets_for_new_rf(schema_ptr s, token_met
         std::unordered_map<sstring, size_t> dc_rep_factor;
         tablet_replica_set replicas = ti.replicas;
 
-        auto tok = token::midpoint(tablets.get_first_token(tb), tablets.get_last_token(tb));
-
-        kqdbg("tb: {}, replicas: {}, last_token: {}", tb, replicas, tok);
-        auto maybeEp = tm->get_endpoint(tablets.get_last_token(tb));
-        if (maybeEp) {
-            locator::host_id ep = *maybeEp;
-
-            dc_rep_factor[*host_id_to_dc(ep)]++;
-            for (auto&& replica : replicas) {
-                dc_rep_factor[*host_id_to_dc(replica.host)]++;
-            }
-        } else {
-            kqdbg("maybeEp == nullptr");
+        for (auto&& replica : replicas) {
+            dc_rep_factor[*host_id_to_dc(replica.host)]++;
```

Comment:
The PR is unreviewable if you change the logic in subsequent commits. Later fixes should be squashed.